src/display/annotation_layer.js

```diff
@@ -305,127 +306,36 @@ var TextAnnotationElement = (function TextAnnotationElementClosure() {
-      var iconName = this.data.name;
+      var image = document.createElement('img');
+      image.style.height = this.container.style.height;
+      image.style.width = this.container.style.width;
       image.src = PDFJS.imageResourcesPath + 'annotation-' +
-        iconName.toLowerCase() + '.svg';
+        this.data.name.toLowerCase() + '.svg';
       image.alt = '[{{type}} Annotation]';
       image.dataset.l10nId = 'text_annotation_type';
-      image.dataset.l10nArgs = JSON.stringify({type: iconName});
-
-      var contentWrapper = document.createElement('div');
-      contentWrapper.className = 'annotTextContentWrapper';
-      contentWrapper.style.left = Math.floor(rect[2] - rect[0] + 5) + 'px';
-      contentWrapper.style.top = '-10px';
-
-      var content = this.content = document.createElement('div');
-      content.className = 'annotTextContent';
-      content.setAttribute('hidden', true);
-
-      var i, ii;
-      if (this.data.hasBgColor && this.data.color) {
-        var color = this.data.color;
-
-        // Enlighten the color (70%).
-        var BACKGROUND_ENLIGHT = 0.7;
-        var r = BACKGROUND_ENLIGHT * (255 - color[0]) + color[0];
-        var g = BACKGROUND_ENLIGHT * (255 - color[1]) + color[1];
-        var b = BACKGROUND_ENLIGHT * (255 - color[2]) + color[2];
-        content.style.backgroundColor = Util.makeCssRgb(r | 0, g | 0, b | 0);
-      }
-
-      var title = document.createElement('h1');
-      var text = document.createElement('p');
-      title.textContent = this.data.title;
-
-      if (!this.data.content && !this.data.title) {
-        content.setAttribute('hidden', true);
-      } else {
-        var e = document.createElement('span');
-        var lines = this.data.content.split(/(?:\r\n?|\n)/);
-        for (i = 0, ii = lines.length; i < ii; ++i) {
-          var line = lines[i];
-          e.appendChild(document.createTextNode(line));
-          if (i < (ii - 1)) {
-            e.appendChild(document.createElement('br'));
-          }
-        }
-        text.appendChild(e);
-
-        image.addEventListener('click', this._toggle.bind(this));
-        image.addEventListener('mouseover', this._show.bind(this, false));
-        image.addEventListener('mouseout', this._hide.bind(this, false));
-        content.addEventListener('click', this._hide.bind(this, true));
-      }
-
-      content.appendChild(title);
-      content.appendChild(text);
-      contentWrapper.appendChild(content);
-      container.appendChild(image);
-      container.appendChild(contentWrapper);
-      return container;
-    },
+      image.dataset.l10nArgs = JSON.stringify({type: this.data.name});
+
+      if (!this.data.popup) {
+        var parameters = {
+          parentContainer: this.container,
+          parentTrigger: image,
+          color: this.data.color,
+          title: this.data.title,
+          contents: this.data.contents
+        };
+        var popupElement = new PopupElement(parameters);
```

Comment:
Done in the new commit, also for the other place where I did this for this commit. Looks better that way.